Code_Exercises/Exercise_02_First_Kernel/README.md

```diff
@@ -0,0 +1,44 @@
+# Exercise 2: First SYCL kernels. Using depencies.
+
+This exercise aims to give an intro to USM functions, single tasks, as well 
+as the concept of data dependencies and synchronization.
+
+ ---
+
+Compile with 
+
+```
+clang++ -fsycl -fsycl-targets=nvptx64-nvidia-cuda solution.cpp
+```
+
+To run:
+```
+SYCL_DEVICE_FILTER=somedevice ./a.out
+```
+---
+
+## Instructions
+
+1. Allocate two ints on device, `a=1` and `b=2`. 
+2. Use a `single_task` to multiply `a` by 2.
+2. Use a separate `single_task` to multiply `b` by 100.
+3. Use a `single_task` to add `a` and `b` together and store the value in `a`.
+4. Memcpy the value back to host and print to stdout.
```

Comment:
We have an instruction to memcpy the data back, but not to memcpy the inputs in.